src/Microsoft.AspNetCore.ApplicationInsights.HostingStartup/ApplicationInsightsStartupLoader.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.ApplicationInsights.AspNetCore;
+using Microsoft.ApplicationInsights.SnapshotCollector;
+
+[assembly: HostingStartup(typeof(Microsoft.AspNetCore.ApplicationInsights.HostingStartup.ApplicationInsightsHostingStartup))]
+
+// To be able to build as <OutputType>Exe</OutputType>
+internal class Program { public static void Main() { } }
+
+namespace Microsoft.AspNetCore.ApplicationInsights.HostingStartup
+{
+    /// <summary>
+    /// A dynamic Application Insights lightup experience
+    /// </summary>
+    public class ApplicationInsightsHostingStartup : IHostingStartup
+    {
+        private const string ApplicationInsightsSettingsFile = "ApplicationInsights.settings.json";
+
+        /// <summary>
+        /// Calls UseApplicationInsights
+        /// </summary>
+        /// <param name="builder"></param>
+        public void Configure(IWebHostBuilder builder)
+        {
+            builder.UseApplicationInsights();
+
+            builder.ConfigureServices(InitializeServices);
+        }
+
+        /// <summary>
+        /// Adds the Javascript <see cref="TagHelperComponent"/> to the <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> associated with the application.</param>
+        private void InitializeServices(IServiceCollection services)
+        {
+            // Add SnapshotCollector telemetry processor.
+            services.AddSingleton<ITelemetryProcessorFactory>(serviceProvider => new SnapshotCollectorTelemetryProcessorFactory(serviceProvider));
+
+            services.AddSingleton<IStartupFilter, ApplicationInsightsLoggerStartupFilter>();
+            services.AddSingleton<ITagHelperComponent, JavaScriptSnippetTagHelperComponent>();
+
+            var home = Environment.GetEnvironmentVariable("HOME");
```

Comment:
wrap into `try{}catch` in case you do not have enough permissions for environment variables

Reply:
Also - do we need logic to make sure it's in fact Azure Web App and not a VS experience (and not any other environment)

Reply:
why would the injection logic be here? the user needs to add ASPNETCORE_HOSTINGSTARTUPASSEMBLIES env var to load this assembly in it's process.

Reply:
HostingStartup is a general purpose module (since the name doesn't suggest it is Azure Web Apps specific). One may decide to use it in container or some custom environment. Thus the logic in it should be generic enough. Any platform-specific logic should be under `if` statement